api/src/controllers/login.js

```diff
@@ -51,6 +51,20 @@ const templates = {
       'privacy.policy'
     ],
   },
+  passwordReset: {
+    file: path.join(__dirname, '..', 'templates', 'login', 'password-reset.html'),
+    translationStrings: [
+      'login',
+      'Password',
+      'privacy.policy'
+    ],
+  }
```

Comment:
I'm debating whether it's better/easier to do this as a separate page, or include it in the initial login form with show/hide based on JS.

I added a comment below that you should not create the session until the password has been changed which adds a bit of complexity because the password reset submission will need to know a) the user, b) the old password, and c) the new password in order to do the password reset and log them in in one go.

Because you need all three pieces of info, and the user has just entered the first two, then that will be lost when you redirect to the new page. However if you just show another form then you can easily access the user and old pass from the hidden form for the second submission.

If you do it as a separate page then you'll have to pass the old password back in clear text which is insecure.

Does that make sense?